devtools/db_import/README.md

```diff
@@ -0,0 +1,421 @@
+  my_pipeline:
+    rules:
+    - filepath_regex: "^(?P<my_arbitrary_capture_name>.*\.json)$" # Any JSON file
+      results: |
+        local getFilepathCapture = function(name) std.parseJson(std.extVar('filepath_captures'))[name];
+        local loadJson = function(filename) std.parseJson(std.native('readFile')(filename));
+
+        local source_json = loadJson(getFilepathCapture('my_arbitrary_capture_name'));
+        /* Let's assume the loaded JSON looks like this:
+        {
+          'benchmark_name' : 'my_arbitrary_benchmark_name',
+          'metrics' : [
+            {
+              'key': '...',
+              'value': 42.42
+            },
+            // ...
+          ]
+        }
+        */
+
+        [
+          {
+            "benchmark_name" : source_json.benchmark_name,
+            "metrics_key" : metric.key,
+            "metrics_value" : metric.value
+          },
+        ] for metric in source_json.metrics
+```
+
+This example defines two boilerplate functions which allow us to read the
+triggering JSON file.  Eventually we read the parsed contents into the
+variable `source_json`. Afterwards we use Jsonnet's
+Array comprehension to create a list of database rows.
+
+## Snippets
+
+The previous example shows that it is sometimes needed to define little
+helper functions in Jsonnet.  To share these between rules or even pipelines
+they can be put into a separate section in the config file and then imported
+from the Jsonnet expression.
+
+This is the previous example, but with the helper functions moved into the
+snippets section:
+
+```yaml
+---
+snippets:
+  getFilepathCapture: function(name) std.parseJson(std.extVar('filepath_captures'))[name]
+  loadJson: function(filename) std.parseJson(std.native('readFile')(filename))
+pipelines: # This config is not complete - some keys are missing
+  my_pipeline:
+    rules:
+    - filepath_regex: "^(?P<my_arbitrary_capture_name>.*\.json)$" # Any JSON file
+      results: |
+        // The import key word can load functions form the snippets section
+        local getFilepathCapture = import "getFilepathCapture";
+        local loadJson = import "loadJson";
+
+        local source_json = loadJson(getFilepathCapture('my_arbitrary_capture_name'));
+
+        [
+          {
+            "benchmark_name" : source_json.benchmark_name,
+            "metrics_key" : metric.key,
+            "metrics_value" : metric.value
+          },
+        ] for metric in source_json.metrics
+```
+
+## Embedding
+
+When Jsonnet expression get larger the YAML config file can become quite
+confusing. That's why we support storing the Jsonnet expression in a separate
+file and embed it into the config file using the `!embed` YAML tag.
+
+Example:
+
+```yaml
+# config.yml
+---
+pipelines: # This config is not complete - some keys are missing
+  my_pipeline:
+    rules:
+    - filepath_regex: "^(?P<my_arbitrary_capture_name>.*\.json)$" # Any JSON file
```

Comment:
It seems like we need to use double backslash here `"^(?P<my_arbitrary_capture_name>.*\\.json)$"`

I tried to run this example but yaml parser seemed to treat this as a incomplete escape character.

Reply:
Oh, you're right. I have the same in my config file. Fixed.